src/mysql_protocol.erl

```diff
@@ -550,7 +550,7 @@ parse_handshake_confirm(<<?MORE_DATA, MoreData/binary>>) ->
 %% prepared statements).
 -spec fetch_response(module(), term(), timeout(), text | binary, [binary()],
                      query_filtermap(), list()) ->
-    {ok, [#ok{} | #resultset{} | #error{}]} | {error, timeout}.
+    {ok, [#ok{} | #resultset{} | #error{}]} | {error, timeout} | {error, closed} .
```

Comment:
Thanks for the explanation! I modified the PR to emit exit, so the reason of closing is clearer there in case of socket closing. Hope this better match your design.

Reply:
Yes, but I'd like the exit to happen in mysql_conn, which is the module responsible for the process. The mysql_protocol module has no concern about processes.

I think the code where we do `exit(timeout)` can be modified to cover `{error, Reason} -> exit(Reason)`.